onnxruntime/wasm/api.h

```diff
@@ -432,6 +432,29 @@ int EMSCRIPTEN_KEEPALIVE OrtTrainingCopyParametersFromBuffer(ort_training_sessio
                                                              size_t parameter_count,
                                                              bool trainable_only);
 
+/**
+ * Gets the input count and output count of the training model associated with the given training handle.
+ * @param traning_handle handle of the traning session
+ * @param input_count [out] a pointer to a size_t variable to accept input_count
+ * @param output_count [out] a pointer to a size_t variable to accept output_count
+ * @returns ORT error code. If not zero, call OrtGetLastError() to get a detailed error message.
+ */
+int EMSCRIPTEN_KEEPALIVE OrtTrainingGetInputOutputCount(ort_training_session_handle_t training_handle,
```

Comment:
Both these new APIs should support eval model as well. You can either create 2 different APIs with model name hard-coded in the api like TrainingModelInputOutputCount or make the model type as an input like isTrainingModel, isEvalModel or something